packages/@lwc/ssr-compiler/src/compile-template/transformers/text.ts

```diff
@@ -9,58 +9,38 @@ import { builders as b, is } from 'estree-toolkit';
 import { esTemplateWithYield } from '../../estemplate';
 import { expressionIrToEs } from '../expression';
 
-import type {
-    Expression as EsExpression,
-    Statement as EsStatement,
-    BlockStatement as EsBlockStatement,
-} from 'estree';
+import { bYieldTextContent, isLastConcatenatedNode } from '../adjacent-text-nodes';
+import type { Statement as EsStatement } from 'estree';
 import type {
     ComplexExpression as IrComplexExpression,
     Expression as IrExpression,
     Literal as IrLiteral,
     Text as IrText,
-    Node as IrNode,
 } from '@lwc/template-compiler';
 import type { Transformer } from '../types';
 
-const bYield = (expr: EsExpression) => b.expressionStatement(b.yieldExpression(expr));
-
-const bYieldEscapedString = esTemplateWithYield`
-    { 
+const bBufferTextContent = esTemplateWithYield`
+    didBufferTextContent = true;
+    {
```

Comment:
```suggestion
    {
        didBufferTextContent = true;
```
In the context of this helper, seems weird to have that outside of the block statement. But in the context of the generated code, I guess it's more weird to have more code than necessary inside our wonky blocks. So I guess the way you wrote it is better.

Thanks for coming to my TED talk.

Reply:
I think we will probably end up refactoring this later anyway. If you look at the generated code, there is a ton of duplication, and Rollup does not inline the blocks as much as I'd like. We may end up just going back to the system of autogenerated variable names for performance reasons.